refactor: unify dom lazy array classes

alandefreitas opened the pull request:
The DomCorpus uses multiple classes that represent lazy arrays. This commit simplifies the dom by unifying all of these array implementations. It's not a new design: it's only consolidating what already exists and removing duplicate code.

sdkrystian commented on include/mrdocs/Metadata/Scope.hpp:
```diff
@@ -29,16 +29,29 @@ namespace mrdocs {
 */
 struct ScopeInfo
 {
-	/** The members of this scope.
```
Unrelated change?

alandefreitas replied:
Mmm... Yes. A small indentation issue. I'll split the commits. :)

vinniefalco commented on include/mrdocs/Metadata/Source.hpp:
```diff
@@ -36,6 +36,15 @@ MRDOCS_DECL
 
+namespace dom {
+    template <>
+    struct ToValue<FileKind>
```
why specialization instead of tag_invoke?

alandefreitas replied:
done :)

alandefreitas committed:
refactor: public settings initialize relativeto

This fixes a warning when using initialization with designators

alandefreitas committed:
refactor(test): remove unused lambda capture

alandefreitas committed:
refactor(LazyObject): header macros belong in LIB

alandefreitas committed:
docs(LazyObject): update documentation

alandefreitas committed:
refactor(DomCorpus): inline operators

alandefreitas committed:
style(Scope): standardize indentation

alandefreitas committed:
docs(Scope): include lookups in documentation

alandefreitas committed:
refactor: unify dom lazy array classes

The DomCorpus uses multiple classes that represented lazy arrays. This commit simplifies the dom by unifying all of these array implementations.

alandefreitas committed:
refactor: dom::Value tag_invoke

alandefreitas committed:
refactor: dom::LazyObject tag_invoke

alandefreitas committed:
refactor: custom ValueFrom for Info types
